src/main/java/seedu/dietmanager/logic/parser/StorageParser.java

```diff
@@ -29,6 +33,25 @@
         return descriptionArray;
     }
 
+    /**
+     * Validates the weightList storage data and parsing it into a valid weight list.
+     *
+     * @param weightListDataLine the weightList data description.
+     * @return weightList in standard form.
+     * @throws NullPointerException if storage data generates a null value.
+     */
+
+    public static List<Double> parseWeightListDataLine(String weightListDataLine)
+            throws NullPointerException, InvalidWeightException {
```

Comment:
May I ask why will this throw NullPointerException? User should not be able to delete all the weights stored, and in that case, there should always still be 1 weight.

Reply:
This is a safeguard in case a an empty String is passed into the function and the parsing function is unable to generate a valid array, when accessing the first value, it will lead to a null value exception